lib/phlex/component.rb

```diff
@@ -34,6 +34,18 @@ def template(...)
     include Node, Context
 
     class << self
+      def register_element(tag_name)
+        unless tag_name.is_a? Symbol
+          raise ArgumentError, "Custom elements must be provided as Symbols"
+        end
```

Comment:
It's a requirement for custom elements to be comprised of two or more words, so we could add a check here.

Reply:
I think they must also be lowercase but we could do a transform to ensure that.